tests/virt_autotest/libvirt_isolated_virtual_network.pm

```diff
@@ -43,26 +43,20 @@ sub run_test {
     upload_logs "vnet_isolated.xml";
     assert_script_run("rm -rf vnet_isolated.xml");
 
-    my $gi_vnet_isolated;
+    my ($mac, $nic);
     foreach my $guest (keys %xen::guests) {
         record_info "$guest", "ISOLATED NETWORK for $guest";
         #figure out that used with virtio as the network device model during
         #attach-interface via virsh worked for all sles guest
-        assert_script_run("virsh attach-interface $guest network vnet_isolated --model virtio --live");
-        #Get the Guest IP Address from ISOLATED NETWORK
-        if (get_var("XEN") || check_var("HOST_HYPERVISOR", "xen")) {
-            my $mac_isolated = script_output("virsh domiflist $guest | grep vnet_isolated | grep -oE \"[[:xdigit:]]{2}(:[[:xdigit:]]{2}){5}\"");
-            script_retry "ip neigh | grep $mac_isolated | grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"", delay => 90, retry => 9, timeout => 90;
-            $gi_vnet_isolated = script_output("ip neigh | grep $mac_isolated | grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"");
-        }
-        else {
-            script_retry "virsh net-dhcp-leases vnet_isolated | grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"", delay => 90, retry => 9, timeout => 90;
-            $gi_vnet_isolated = script_output("virsh net-dhcp-leases vnet_isolated | grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"");
-        }
+        $mac = '00:16:3e:32:' . (int(rand(89)) + 10) . ':' . (int(rand(89)) + 10);
+        assert_script_run("virsh attach-interface $guest network vnet_isolated --model virtio --mac $mac --live", 60);
+        $nic = script_output "ssh root\@$guest ls /sys/class/net/ | grep -v 'lo\\|eth0'";
+        assert_script_run("ssh root\@$guest ip a a 192.168.122.207/24 dev $nic");
```

Comment:
ISOLATED NETWORK will provides a completely isolated private network. so, it should not add the isolated private network for guest manually. we should let guest system get the isolated private IP address from isolated private network - we need to confirm this workload for isolated network mode.

Reply:
I switched to DHCP, can you check the `test_network_interface()` function in `lib/virt_autotest/virtual_network_utils.pm` file?